### PR TITLE
topology2: intel: dmic-generic: Simplify the core ID setting

### PR DESCRIPTION
**Pull request by ranj063**
When all components of a pipeline are scheduled on the same core, use the pipeline's core_id attribute to set the core_id for all the components in it. Also, remove the default core_id in the module-copier and eqiir class definitions so that it doesn't take priority over the value passed from the pipeline object.

### CODE REVIEW

File: tools/topology/topology2/platform/intel/dmic-generic.conf [hunk 1]
```diff
@@ -76,10 +76,9 @@ IncludeByKey.PASSTHROUGH {
 			{
 				format		$FORMAT
 				index 		$DMIC0_HOST_PIPELINE_ID
-
+				core_id		$DMIC_CORE_ID
 				Object.Widget.host-copier.1 {
 					stream_name $DMIC0_PCM_CAPS
-					core_id $DMIC_CORE_ID
```
Comment (plbossart) [inline tools/topology/topology2/platform/intel/dmic-generic.conf:82]: @ranj063 for my education, how does the pipeline-level core_id get mirrored in each module instance? is this something done by the topology parser or something done by the driver+firmware?

Comment (ranj063) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: @plbossart it done by the topology compiler. All children objects automatically inherit their parent object's attribute values unless they have them set explicitly too.

Comment (plbossart) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: what happens if the base class for an object has a different value to that of the object parent? Who wins? The parent or the base class?

Comment (ranj063) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: Base class

Comment (ranj063) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: The topology2 documentation has the information on the order of priority for where the value is inferred from.

Comment (plbossart) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: I am sure everyone has read that documentation, right? I wouldn't even know where to find a published version.

**Review by juimonen — APPROVED** [review]
lgtm, although touching many other files than dmic-generic (in the commit title)

**Review by yongzhi1 — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Review by kv2019i — CHANGES_REQUESTED** [review]
All dmic capture tests fail in CI, please check.

**Comment by ranj063** [comment]
> All dmic capture tests fail in CI, please check.

@kv2019i i was testing with core ID 222 just be able to spot it easily in the conf file and lef it in by mistake. FIxed now. tests should pass

File: tools/topology/topology2/platform/intel/dmic-generic.conf (hunk 1 above)
Comment (ranj063) [reply tools/topology/topology2/platform/intel/dmic-generic.conf:82]: @plbossart there's preliminary documentation for topology2but we're going to enhance it to include all recently introduced features. Please stay tuned.

**Comment by kv2019i** [comment]
Apparently we did not have 200+ cores. ;) That was a good smoke test. This version now passes tests. Only known PM fail in https://sof-ci.01.org/sofpr/PR7712/build8776/devicetest/index.html , proceeding with merge.

**Review by kv2019i — APPROVED** [review]

**Pull request merged**